cdap-client/src/main/java/co/cask/cdap/client/config/ConnectionConfig.java

```diff
@@ -66,10 +67,11 @@ public ConnectionConfig(Id.Namespace namespace, String hostname, int port, boole
     this.hostname = hostname;
     this.port = port;
     this.sslEnabled = sslEnabled;
+    this.username = System.getProperty("user.name");
```

Comment:
I was thinking whatever the user is authenticated as. For example if you login to a secure CDAP cluster as the user "sdf", you'd see "cdap(http://sdf@host:port)> " as the prompt. It may be tricky to implement since the existing CDAP authentication server doesn't really tell you who you are (and it may not make sense to have a username in some cases, e.g. when using a different authentication scheme that doesn't accept simple username/password).